### DISCUSSION TITLE
Virtual Button for Zigbee Devices

### DISCUSSION THREAD

**Discussion by drtorchwood** [arendst/Tasmota#14208] [Q&A] [answered]
Hi all, 

I'm struggeling with the following problem: Tasmota (v 10.0.0) is running on a wemos 1 mini with a CC2530 for Zigbee. There are some more devices (botton, relay, sensors) and now there is no GPIO left. Via Zigbee, three plugs are connected. I was wondering if I can create some kind of virtual buttons and map them to the devices. Then, I could control them in the GUI and for the mqtt server they would appear like "normal" tasmota relays, e.g. I could send something like `Power1 ON`.
A slightly different solution would be to have a relay in the GUI (but not connected to any gpio) and then use the rules to connect the on/off commands to Zbsend commands. 

Any idea for this setup? 
Also any insight how others use zigbee plugs in their system is appreciated. I'm looking for a procedure that is flexible and can easily be extended (perhaps even via node-red putting ZbSend commands in mqtt messages). 

Thanks,
Jan

**Comment by s-hadinger** [comment, marked as answer]
Tasmota is not designed to do this, and maybe you can do sort of it with Rules, but it may be fragile.

A better option would be to switch to ESP32 and do some Berry scripting, which gives you many options to customize the behavior.
